packages/react-dev-utils/webpackHotDevClient.js

```diff
@@ -288,5 +287,18 @@ function tryApplyUpdates(onHotUpdateSuccess) {
       // While we were updating, there was a new update! Do it again.
       tryApplyUpdates();
     }
-  });
+  }
+
+  // https://webpack.github.io/docs/hot-module-replacement.html#check
+  var result = module.hot.check(/* autoApply */true, checkCallback);
+
+  // webpack 2 support
+  if (result && result.then) {
+    result.then(
+      function(updatedModules) {
+        checkCallback(null, updatedModules);
+      },
+      checkCallback
```

Comment:
For explicitness let’s also write this as a function that passes error as the first argument and `null` as a second one.